src/kili/services/plugins/upload.py

```diff
@@ -214,18 +218,26 @@ def _retrieve_upload_url(self, is_updating_plugin: bool) -> str:
         upload_url = format_result("data", result)
         return upload_url
 
-    @staticmethod
-    def _create_zip(script_path: Path, requirements_path: Optional[Path], tmp_directory: Path):
+    def _create_zip(
+        self, file_paths: List[Path], requirements_path: Optional[Path], tmp_directory: Path
+    ):
         """
         Create a zip file from python file and requirements.txt
         (if user has defined a path to a requirements.txt)
         """
         zip_path = tmp_directory / "archive.zip"
 
         with ZipFile(zip_path, "w") as archive:
-            archive.write(script_path, "main.py")
+
+            if len(file_paths) == 0:
+                archive.write(file_paths[0], "main.py")
+
+            else:
+                for path in file_paths:
+                    archive.write(path, path.relative_to(self.plugin_path))
+
             if requirements_path:
-                archive.write(requirements_path, "requirements.txt")
+                archive.write(requirements_path, requirements_path.relative_to(self.plugin_path))
```

Comment:
Why does it need to be relative and not just at the root of the zip ?

Reply:
It's because if I have a folder like that:
- plugin-folder
   - main.py
   - requirements.txt
   - sub-folder
       - helpers.py
       
The value of `requirements_path` would be `plugin-folder/requirements.txt` while `requirements_path.relative_to(self.plugin_path) = 'requirements.txt'`

However, this line change is useless. I did `archive.write(requirements_path, "requirements.txt")` today on another PR to fix the upload of requirements.txt while I did `archive.write(requirements_path, requirements_path.relative_to(self.plugin_path))` yesterday. Both way are doing the same thing